### PR TITLE
add optional dhcp daemonset to rke2-multus

### PR DESCRIPTION
**Pull request by thomasferrandiz**
This PR adds an optional daemonSet to the multus chart.
It is required to use the dhcp ipam plugin with multus.

### CODE REVIEW

File: packages/rke2-multus/charts/templates/dhcp-daemonSet.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
```
Comment (mgfritch) [inline packages/rke2-multus/charts/templates/dhcp-daemonSet.yaml:1]:
```suggestion
```
remove?

File: packages/rke2-multus/charts/templates/dhcp-daemonSet.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+{{- if .Values.manifests.dhcpDaemonSet }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: multus-dhcp-ds
```
Comment (mgfritch) [inline packages/rke2-multus/charts/templates/dhcp-daemonSet.yaml:6]:
```suggestion
  name: {{ .Release.Name }}-ds
```
(nit) add release name ?

Comment (brandond) [reply packages/rke2-multus/charts/templates/dhcp-daemonSet.yaml:6]: I would also nit on putting -ds on the end; we know it's a daemonset. There's no need to use hungarian notation here.

Comment (thomasferrandiz) [reply packages/rke2-multus/charts/templates/dhcp-daemonSet.yaml]: done!

**Review by manuelbuil — APPROVED** [review]

**Review by brandond — APPROVED** [review]

**Pull request merged**